File: tasks/translation/output/Qwen/Qwen2.5-Coder-32B-Instruct-AWQ/codenet/direct/Java/Go/s893190600_commented.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

func main() {
	in := bufio.NewReader(os.Stdin)
	out := bufio.NewWriter(os.Stdout)
	defer out.Flush()

	solver := Solver{in: in, out: out}
	solver.solve()
}

type Solver struct {
	in  *bufio.Reader
	out *bufio.Writer
}

func (s *Solver) solve() {
	H := s.nextInt()
	W := s.nextInt()
	C := make([][]rune, H+2)
	for i := range C {
		C[i] = make([]rune, W+2)
	}
	for i := 0; i < H; i++ {
		line := s.nextString()
		for j, c := range line {
			C[i+1][j+1] = c
		}
	}
	if H == 1 && W == 1 {
		s.println("No")
		return
	}
	for i := 1; i <= H; i++ {
		for j := 1; j <= W; j++ {
			if C[i][j] == '#' && C[i-1][j] != '#' && C[i+1][j] != '#' &&
				C[i][j-1] != '#' && C[i][j+1] != '#' {
				s.println("No")
				return
			}
		}
	}
	s.println("Yes")
}

func (s *Solver) nextInt() int {
	line := s.nextString()
	i, _ := strconv.Atoi(line)
	return i
}

func (s *Solver) nextLong() int64 {
	line := s.nextString()
	i, _ := strconv.ParseInt(line, 10, 64)
	return i
}

func (s *Solver) nextDouble() float64 {
	line := s.nextString()
	i, _ := strconv.ParseFloat(line, 64)
	return i
}

func (s *Solver) nextString() string {
	for {
		line, _, err := s.in.ReadLine()
		if err != nil {
			panic(err)
		}
		str := strings.TrimSpace(string(line))
		if len(str) > 0 {
			return str
		}
	}
}

func (s *Solver) println(a ...interface{}) {
	fmt.Fprintln(s.out, a...)
}

func (s *Solver) print(a ...interface{}) {
	fmt.Fprint(s.out, a...)
}

// End of Code

File: tasks/translation/output/Qwen/Qwen2.5-32B-Instruct-AWQ/codenet/direct/comment/uncomment/direct/Java/Python/s893190600_commented.py
import sys
from collections import defaultdict, deque

class Solver:
    def __init__(self, input_stream, output_stream):
        self.inp = input_stream
        self.out = output_stream

    def solve(self):
        H = self.ni()
        W = self.ni()
        C = [['.' for _ in range(W + 2)] for _ in range(H + 2)]
        for i in range(1, H + 1):
            line = self.ns()
            for j in range(1, W + 1):
                C[i][j] = line[j - 1]
        
        if H == 1 and W == 1:
            self.prn("No")
            return
        
        for i in range(1, H + 1):
            for j in range(1, W + 1):
                if C[i][j] == '#' and C[i-1][j] != '#' and C[i+1][j] != '#' and C[i][j-1] != '#' and C[i][j+1] != '#':
                    self.prn("No")
                    return
        
        self.prn("Yes")

    def prn(self, s):
        print(s, file=self.out)

    def ni(self):
        return int(self.inp.readline())

    def nl(self):
        return int(self.inp.readline())

    def nd(self):
        return float(self.inp.readline())

    def ns(self):
        return self.inp.readline().strip()

def main():
    input_stream = sys.stdin
    output_stream = sys.stdout
    solver = Solver(input_stream, output_stream)
    solver.solve()

if __name__ == "__main__":
    main()

# End of Code
